src/writer/basic.rs

```diff
@@ -42,8 +43,8 @@ pub struct Basic {
     /// Current indentation that events are outputted with.
     indent: usize,
 
-    /// Indicates whether last lines should be cleared.
-    needs_clear: bool,
+    /// Number of lines to clear, if any.
+    lines_to_clear: Option<usize>,
```

Comment:
Just `usize` is OK. If we have `0`, then we don't clear. No need for `Option` here.